registry/server/tagstorage_zk.go

```diff
@@ -166,6 +167,62 @@ func (t *ZKTagStorage) GetTags(o KafkaObject) (TagSet, error) {
 	return tags, nil
 }
 
+// GetAllTags returns all tags stored in the tagstore, keyed by the resource they correspond to.
+func (t *ZKTagStorage) GetAllTags() (map[KafkaObject]TagSet, error) {
+
+	tags := map[KafkaObject]TagSet{}
+
+	brokerTags, err := t.GetAllTagsForType("broker")
+	if err != nil {
+		return nil, err
+	}
+	for broker, brokerTags := range brokerTags {
+		tags[broker] = brokerTags
+	}
+
+	topicTags, err := t.GetAllTagsForType("topic")
+	if err != nil {
+		return nil, err
+	}
+	for topic, topicTags := range topicTags {
+		tags[topic] = topicTags
+	}
+
+	return tags, nil
+}
+
+// GetAllTagsForType gets all the tags for objects of the given type. A convenience method that makes getting every tag a little easier.
+func (t *ZKTagStorage) GetAllTagsForType(kafkaObjectType string) (map[KafkaObject]TagSet, error) {
+	zNode := fmt.Sprintf("/%s/%s/", t.Prefix, kafkaObjectType)
```

Comment:
I can't recall from memory, but if we don't have a prefix, will we get `//path` (double slashes) here?

Reply:
We will, but an empty string is an invalid prefix, at tagstorage_zk.go ln 23

Reply:
This is fine, really; if anything we should just enforce a non "" Registry prefix anyway at startup. I can't imagine anyone wanting its storage blasted into the root znode.